core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,8 +62,14 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // visible for testing
+  private[spark] val isPartitionEvalParallel: Boolean =
+    rdds.length > conf.getInt("spark.rdd.parallelListingThreshold", 10)
```

Comment:
Changing a config option is turning on a different codepath; in theory it should go through a QE process. At least with a config option, when it does play up, you can roll back by changing the switch and restarting: you've saved the cost of redeploying the old code.

Undocumented options are dangerous though

* nobody in support will know what to do if problems surface and the switch is set
* someone is going to find it in the code and wonder "what does this do"... the origins will be lost in the folklore of the project.

What about having a section in the docs on "unstable+unsupported options", list it with a link to the JIRA and a "use at your own risk" disclaimer?